validator/client/src/test/java/tech/pegasys/artemis/validator/client/signer/SignerTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.validator.client.signer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.primitives.UnsignedLong;
+import org.apache.tuweni.bytes.Bytes;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
+import tech.pegasys.artemis.datastructures.operations.AttestationData;
+import tech.pegasys.artemis.datastructures.state.Fork;
+import tech.pegasys.artemis.datastructures.util.DataStructureUtil;
+import tech.pegasys.artemis.datastructures.validator.MessageSignerService;
+import tech.pegasys.artemis.util.async.SafeFuture;
+import tech.pegasys.artemis.util.bls.BLSSignature;
+
+class SignerTest {
+
+  private final MessageSignerService signerService = mock(MessageSignerService.class);
+  private final DataStructureUtil dataStructureUtil = new DataStructureUtil();
+  private final Fork fork = dataStructureUtil.randomFork();
+
+  private final Signer signer = new Signer(signerService);
+
+  @Test
+  public void shouldCreateRandaoReveal() {
+    final BLSSignature signature = dataStructureUtil.randomSignature();
+    final Bytes expectedSigningRoot =
+        Bytes.fromHexString("0x84c8825ccb169a0d1109ff2b96185ea2bd6f66ed198f2e322473773fff3ed91b");
+    when(signerService.signRandaoReveal(expectedSigningRoot))
+        .thenReturn(SafeFuture.completedFuture(signature));
+    final SafeFuture<BLSSignature> reveal =
+        signer.createRandaoReveal(UnsignedLong.valueOf(7), fork);
```

Comment:
I don't have a suggestion, however, it took me a good amount of time to infer (still not sure) that passing epoch number 7 and the given fork value results in the expectedSigningRoot that you declare above. I wish there was a more clear way of showing that.

Reply:
Yeah, ultimately those tests are really checking that the signing root was calculated correctly and that it was delegated to the right method on the `MessageSignerService`. I can't actually see a way to simplify them either.  We could have a separate object responsible for calculating the signing root and test that separately but it just makes the production code more complex and I don't think it's worth it.